lib/new_post_manager.rb

```diff
@@ -318,4 +318,18 @@ def perform_create_post
 
     result
   end
+
+  private
+
+  def self.allowed_groups
+    if SiteSetting.approve_unless_allowed_groups.is_a?(String)
+      SiteSetting.approve_unless_allowed_groups.split("|").map(&:to_i)
+    else
+      [SiteSetting.approve_unless_allowed_groups]
+    end
+  end
+
+  def self.approve_unless_allowed_groups_is_trust_level_0
+    SiteSetting.approve_unless_allowed_groups == Group::AUTO_GROUPS[:trust_level_0]
```

Comment:
I think this would need to be `SiteSetting.approve_unless_allowed_groups_map.includes?(Group::AUTO_GROUPS[:trust_level_0])`, then you don't need the clunky separate method either.